src/main/java/formflow/library/UploadController.java

```diff
@@ -168,17 +168,19 @@ RedirectView delete(
               // - remove things after .ext
               // - have '-thumbnail' before the .txt
               // - add txt as extension
+              // TODO - maybe this should be replaceLast, so if there are other '.' in the string we don't run into issues?
```

Comment:
@bseeger is there a `replaceLast`? I only see `replace` and `replaceAll` in my autocomplete?

Reply:
We also control the filename, so if there's another `.` it would be because we add it to the cloud filename.

Reply:
All this being said, the current regex here doesn't work 😢 

<img width="1015" alt="Screenshot 2022-12-01 at 5 45 37 PM" src="https://user-images.githubusercontent.com/9101728/205182597-42340ff8-dad5-4ae6-b4be-98769a34e96c.png">

Reply:
I had to use `.replaceFirst` as it would allow regex. `.replace()` does not accept regex to find a target. Only `.replaceFirst` and `.replaceAll`.